tests/structs/test_query_output_details.py

```diff
@@ -15,37 +15,47 @@ def setUp(self) -> None:
 
     @parameterized.expand(
         [
-            ("no args use defaults", False, None),
-            ("properties given, not output type", True, None),
-            ("no properties given, output type given", False, "TO_OBJECT_LIST"),
-            ("properties and output type given", True, "TO_LIST"),
+            ("none given", {}),
+            ("props given", {"properties_to_select": ["server_id", "server_name"]}),
+            ("output type given", {"output_type": "TO_LIST"}),
+            ("group by given", {"group_by": "server_name"}),
+            ("sort by given", {"sort_by": ["server_name", "server_id"]}),
         ]
     )
-    def test_from_kwargs(self, _, set_mock_props, mock_output_type):
+    def test_from_kwargs(self, _, mock_kwargs):
```

Comment:
Don't use parametrized to name test cases, instead move the boilerplate to a helper private method and create multiple test cases for it instead

```
def test_from_kwargs_group_by_given(self):
   self._run_from_kwargs_case({"group_by": "server_name"})

def test_from_kwargs_sort_given(self):
   self._run_from_kwargs_case({"sort_by": ["server_name", "server_id"]})
```

There's a few benefits:
- Easier to debug individual test cases
- Easier to read than following params through parametrized 
- The CLI will print the proper name for the test 
- and that name you can easily grep for as entire string, rather than a sub-string